zetaclient/bitcoin_client.go

```diff
@@ -623,23 +637,69 @@ func (ob *BitcoinChainClient) fetchUTXOS() error {
 	return nil
 }
 
+// This func tries increasing the artificial pending nonce of outTx (if lagged behind).
+// There could be many (unpredictable) reasons for a pending nonce lagging behind, for example:
+// 1. The zetaclient gets restarted.
+// 2. The tracker is missing in zetacore.
+func (ob *BitcoinChainClient) refreshPendingNonce() {
+	// get pending nonces from zetacore
+	p, err := ob.zetaClient.GetPendingNoncesByChain(ob.chain.ChainId)
+	if err != nil {
+		ob.logger.ChainLogger.Error().Err(err).Msg("refreshPendingNonce: error getting pending nonces")
+	}
+
+	// increase pending nonce if lagged behind
+	ob.mu.Lock()
+	pendingNonce := ob.pendingNonce
+	ob.mu.Unlock()
+
+	if p.NonceLow > 0 && uint64(p.NonceLow) >= pendingNonce {
+		// get the last included outTx hash
+		txid, err := ob.getOutTxidByNonce(uint64(p.NonceLow)-1, false)
+		if err != nil {
+			ob.logger.ChainLogger.Error().Err(err).Msg("refreshPendingNonce: error getting last outTx txid")
+		}
```

Comment:
You're right. This refreshing tries to update the pending nonce without caring about errors (it will succeed sooner or later). As I put in the `refreshPendingNonce` func's comment, there could be many (unpredictable) reasons for a pending nonce lagging behind, for example:
1. The zetaclient gets restarted.
2. The tracker is missing in zetacore.
![image](https://github.com/zeta-chain/node/assets/34498985/0e11d495-7941-4fe6-95dc-cec02ea0f459)